### DISCUSSION TITLE
Concept of MetricLevel [new feature?]

### DISCUSSION THREAD

**Discussion by mcfongtw** [dropwizard/metrics#2756] [Ideas]
Has there been any discussion about introducing a concept of Metric Level to each Metric instance?

Similar to log level in a logging framework, a metric level defines when a metric is needed. User can (statically / dynamically) enable / disable a metric according to the verbose level that is set in the reporter, i.e. the default level in reporter is set at INFO level and user can opt-in to DEBUG level if additional metric data is needed.

**Comment by joschi** [comment]
@mcfongtw Is there any "prior art" to this idea?

Do other metrics libraries support a concept of "Metric Levels" already?

**Reply by mcfongtw** [reply]
I can tell the similar [MetricLevel](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/metrics/MetricLevel.html) concept is used for internal metrics in AWS SDK. From my past experience using metrics, we had thousands of metrics, from mission-critical metrics to ones only for troubleshooting.  It would be nice to  have a flexible way to adjust the verbosity of metric reported from an application.
